### PR TITLE
refactor(levm): ef test runner

### PR DESCRIPTION
**Pull request by ilitteri**
**Description**

- Fixes EF tests parsing
- Adds `levm_runner` module
- Adds `revm_runner` module
- Refactors `runner` module
- Fixes post-state validations
- Refactors `report` module.
- Adds more reports.

### CODE REVIEW

**Review by maximopalopoli — APPROVED** [review]
LGTM after passing the CI

File: cmd/ef_tests/levm/runner/levm_runner.rs [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+                    return Err(EFTestRunnerError::FailedToEnsurePostState(
+                        execution_report.clone(),
+                        error_reason,
+                    ));
+                }
+                // Execution result was successful and no exception was expected.
+                None => {
+                    let levm_account_updates = get_state_transitions(execution_report);
+                    let pos_state_root = post_state_root(&levm_account_updates, test);
+                    let expected_post_state_value = test.post.iter().next().cloned();
```
Comment (ilitteri) [inline cmd/ef_tests/levm/runner/levm_runner.rs:175]: Fix

File: cmd/ef_tests/levm/deserialize.rs [hunk 2]
```diff
@@ -122,103 +122,104 @@ where
...
-            })?,
-            transactions,
-        };
-        Ok(ef_test)
+            })?;
+
+            let mut transactions = HashMap::new();
+
+            // Note that inthis order of iteration, in an example tx with 2 datas, 2 gasLimit and 2 values, order would be
+            // 111, 112, 121, 122, 211, 212, 221, 222
```
Comment (maximopalopoli) [inline cmd/ef_tests/levm/deserialize.rs:154]: I think we can delete this comment since the `tx_id` notion is not used anymore

**Review by maximopalopoli — APPROVED** [review]
LGTM

File: cmd/ef_tests/levm/deserialize.rs (hunk 2 above)
Comment (ilitteri) [reply cmd/ef_tests/levm/deserialize.rs:154]: Yes, let's remove it in a cleanup PR

**Review by JereSalo — APPROVED** [review]
LGTM. It would be very good if we could run tests with `--nocapture` for debugging purposes but I don't know if clap allows that.

**Pull request merged**